src/user-safe/UserSafe.sol

```diff
@@ -373,6 +416,41 @@ contract UserSafe is IUserSafe, Initializable, OwnableUpgradeable {
         emit WithdrawalProcessed(tokens, amounts, recipient);
     }
 
+    /**
+     * @inheritdoc IUserSafe
+     */
+    function toggleRecovery() external onlyOwner {
```

Comment:
I think, instead of doing `toggle`, it would be better to set it explicitly `_isRecoveryActive` = true/false by a setter